airflow/auth/managers/base_auth_manager.py

```diff
@@ -457,14 +457,28 @@ def register_views(self) -> None:
         """Register views specific to the auth manager."""
 
     @staticmethod
-    def _get_token_signer():
+    def _get_token_signer() -> JWTGenerator:
         """
         Return the signer used to sign JWT token.
 
         :meta private:
         """
-        return JWTSigner(
-            secret_key=get_signing_key("api", "auth_jwt_secret"),
-            expiration_time_in_seconds=conf.getint("api", "auth_jwt_expiration_time"),
+        return JWTGenerator(
+            secret_key=get_signing_key("api_auth", "jwt_secret"),
+            valid_for=conf.getint("api", "auth_jwt_expiration_time"),
+            audience="front-apis",
+        )
+
+    @staticmethod
+    def _get_token_validator() -> JWTValidator:
+        """
+        Return the signer used to sign JWT token.
+
+        :meta private:
+        """
+        return JWTValidator(
+            # issuer=conf.get("api_auth", "jwt_iussuer"),
+            secret_key=get_signing_key("api_auth", "jwt_secret"),
+            leeway=conf.getint("api", "auth_jwt_expiration_time"),
```

Comment:
`leeway` should not come from the expiration time. It's to ensure validity based on small clock offsets and deltas.

Reply:
Good catch, agreed. (It _did_ before, there was just a single thing but yeah that is wrong)